src/lecture_feedback/gui.py

```diff
@@ -32,43 +29,36 @@ def show_room_selection_screen(
                 st.warning("Please enter a Room ID to join.")
             else:
                 try:
-                    application_state.join_room(room_id, session_id)
+                    app.join_room(room_id)
                     st.rerun()
                 except ValueError:
                     st.error("Room ID not found")
 
 
-def show_active_room(room: Room, session_id: str) -> None:
+def show_active_room(app: App, room: Room) -> None:
     st.title("Active Room")
     st.write(f"**Room ID:** `{room.room_id}`")
     st.divider()
 
     if st.button(UserStatus.RED.value):
-        room.set_session_status(session_id, UserStatus.RED)
+        app.set_session_status(UserStatus.RED)
     if st.button(UserStatus.YELLOW.value):
-        room.set_session_status(session_id, UserStatus.YELLOW)
+        app.set_session_status(UserStatus.YELLOW)
     if st.button(UserStatus.GREEN.value):
-        room.set_session_status(session_id, UserStatus.GREEN)
+        app.set_session_status(UserStatus.GREEN)
     if st.button(UserStatus.UNKNOWN.value):
-        room.set_session_status(session_id, UserStatus.UNKNOWN)
+        app.set_session_status(UserStatus.UNKNOWN)
 
     for sid, status in room:
         st.write(f"Session {sid}: {status.value}")
 
 
-@st.cache_resource
-def get_application_state() -> ApplicationState:
-    return ApplicationState()
-
-
 def run() -> None:
     st_autorefresh(interval=2000, key="data_refresh")
 
-    application_state = get_application_state()
-    session_state = SessionState()
-    session_id = session_state.session_id
+    app = App()
 
-    if (room := application_state.get_session_room(session_id)) is not None:
-        show_active_room(room, session_id)
+    if (room := app.get_active_room()) is not None:
```

Comment:
I think room shouldnt be a local variable, everthing should be handled by the facade / app

Reply:
That was my first approach as well, but you still need to display room contents (room id, session id, user status) in `show_active_room`. And if you only there get the room or its contents, you need to awkwardly check if you're in a room or not.